numpy/lib/_function_base_impl.py

```diff
@@ -415,14 +415,13 @@ def average(a, axis=None, weights=None, returned=False, *,
     weights : array_like, optional
         An array of weights associated with the values in `a`. Each value in
         `a` contributes to the average according to its associated weight.
-        The weights array can either be 1-D (in which case its length must be
-        the size of `a` along the given axis) or of the same shape as `a`.
+        The array of weights must be the same shape as `a` if no axis is
+        specified, otherwise the weights must have dimensions and shape
+        consistent with `a` along the specified axis..
         If `weights=None`, then all data in `a` are assumed to have a
-        weight equal to one.  The 1-D calculation is::
-
-            avg = sum(a * weights) / sum(weights)
```

Comment:
@mhvk I didn't really get what was there before or how it was related to "The 1-D calculation" specifically, and it seemed obvious to me how the average would be defined, so I removed it. Very happy to add it back as you suggested though, with the minor tweak `where the sum is over all included elements` ('elements' of the array [rather than 'points'] are referred to elsewhere in the docstring).

Reply:
I'm a bit worried about the blank lines that are now removed - sphinx is super picky about that and it might get typeset wrong. Just to be sure, maybe change to
```
        weight equal to one. The calculation is::

            avg = sum(a * weights) / sum(weights)

        where the sum is over all included elements.
```

Reply:
@mhvk Thanks and apologies - good spot. Indeed I think sphinx might have got upset by that.